lib/view_model/trade_details_view_model.dart

```diff
@@ -90,6 +91,17 @@ abstract class TradeDetailsViewModelBase with Store {
           value: trade.provider.toString()));
     }
 
+    if (trade.provider is ChangeNowExchangeProvider) {
```

Comment:
It will never works, this statement will be always false. `provider` is not sub class (or implements) `ExchangeProvider` or `ChangeNowExchangeProvider`.